kernel/src/table_features/mod.rs

```diff
@@ -41,12 +41,7 @@ mod timestamp_ntz;
 #[serde(rename_all = "camelCase")]
 #[internal_api]
 pub(crate) enum TableFeature {
-    /// CatalogManaged tables:
-    /// <https://github.com/delta-io/delta/blob/master/protocol_rfcs/catalog-managed.md>
-    CatalogManaged,
-    #[strum(serialize = "catalogOwned-preview")]
-    #[serde(rename = "catalogOwned-preview")]
-    CatalogOwnedPreview,
+    // Writer-only features
```

Comment:
I like this. Small nit to distinguish the sections:
```suggestion
     /////////////////////////
    // Writer-only features //
    //////////////////////////
```